hardware/detect.py

```diff
@@ -990,7 +944,9 @@ def main():
     args = parse_args(sys.argv[1:])
 
     hrdw = []
-    detect_areca(hrdw)
+    areca_hrdw = areca.detect()
```

Comment:
I like the change, this is clearly a good idea for the other modules too.

That said I'm not a big fan of having to process an if areca_hrdw after the call. Can we keep a similar way to add elements into the hrdw structure ?

Reply:
The idea is exactly give a base for a change in the other modules too, so I'm glad it got some approval :)

Consider that in the end the structure here will be very different, I would like to stop passing the hrdw structure all around the modules, that's one of the reasons why I started this.
For the moment we can condense the three lines into one, or we can remove the if and just append the return value, it should not be a problem, you will get a None inside the final list, but we can filter that out.

Or you had a different solution in mind?